### PR TITLE
Update Global Sidebar

### PR DESCRIPTION
**Pull request by eoigal**
Fixes https://github.com/Automattic/dotcom-forge/issues/5549

This PR will add a global sidebar to the top level (L1) sites dashboard.

Related to Design prototype here - pe7F0s-1zX-p2

### Overview

This is a breakdown of global sidebar;

**Sites Dashboard**
* Clicking on the 'WordPress.com' logo or back link will bring the user back to the sites dashboard
sites-dashboard

**Manage Domains**
* Available from the `Domains` sidebar menu in L1
domain-manage

**Reader**
* Available from the `Reader` sidebar menu in L1
reader

**Profile (Me)**
* Available from the Profile user info link in the sidebar footer menu in L1
profile

**Notification Panel**
* Available from the `Notifications` sidebar menu in L1
* The panel is still opening on left, may want to switch direction (waiting for feedback first).
notification-panel

**Help Center**
* Available from clicking the `?` icon in the sidebar footer menu in L1
help-center

**Command Palette (search)**
* Available from clicking the search icon in the sidebar header menu in L1
command-palette

### Testing
Apply patch and confirm works along with the latest prototype behaviour and layout

### Things to do/figure out

- [ ] Update styles smaller resolutions - haven't accounted yet for smaller screens yet
- [ ] Need to improve feature gating so we can deploy safely
- [ ] Will need to change how we show/hide the command palette after that component has been refactored - paYJgx-4Cq-p2
- [ ] Should we show the notification panel on the left or right? - should we skip this and try to render them like we do on Reader?
- [ ] Fix bug where sidebar renders out of alignment and incorrect dimensions (think related to scroll/resize event)

### CODE REVIEW

**Review by Addison-Stavlo — COMMENTED** [review]
I had some build issues as noted in slack: p1707512802239769/1707502189.792039-slack-C03NLNTPZ2T

The global level and navigation within the reader seems great. Two issues I am having once I select a site and get out of the global level:
* The sidebar is still showing global level content, I can't get anywhere with respect to managing the site.
* The my home page is a bit off the top of the screen.

global-nav-1

**Comment by roo2** [comment]
>  Should we show the notification panel on the left or right? - should we skip this and try to render them like we do on Reader?

I asked for clarity here p1707697121478539/1707303699.626449-slack-C06F5U4BU83

**Comment by roo2** [comment]
I think this is going to be pretty tricky to get behind a feature toggle!

**Comment by roo2** [comment]
Rather than add a bunch of visual feedback, I worked on the CSS directly and pushed some commits to this PR to make it more like the prototype. I hope that's useful and isn't stepping on your toes too much, let me know if I can help better integrate it.

Prototype            | Before | After
:-------------------------:|:-------------------------:|:-------------------------:
Screenshot 2024-02-12 at 12 22 21 pm |  Screenshot 2024-02-12 at 10 22 13 am | Screenshot 2024-02-12 at 12 22 13 pm



----

The CSS affected reader too, I tidied it up a bit so it is not too broken, but it looks pretty clunky to me, I think we will need some detailed designs of the reader menu including the sub-menus etc. This is how reader menu is looking at the moment. I think it would look better if we hid the separators:

Current | Separators shown
:-------------------------:|:-------------------------:
Screenshot 2024-02-12 at 12 13 10 pm | Screenshot 2024-02-12 at 12 20 49 pm

**Comment by lsl** [comment]
Can this be deployed behind the feature flag? I want to start working on the L2 changes (Global Site View)

Not too sure about these changes https://github.com/Automattic/wp-calypso/pull/86766#issuecomment-1937991969 I thought we wanted some newer wp.com design or something? or is that coming later? @davemart-in @roo2

File: client/layout/global-sidebar/hooks/use-global-sidebar.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const useGlobalSidebar = ( siteId, sectionGroup ) => {
+	// TODO: need to expand on logic here - could add feature flag or some environment variable to control this
+	const shouldShowGlobalSidebar =
+		sectionGroup === 'me' ||
+		sectionGroup === 'reader' ||
+		sectionGroup === 'sites-dashboard' ||
+		( sectionGroup === 'sites' && ! siteId );
+	return { shouldShowGlobalSidebar };
+};
```
Comment (lsl) [inline client/layout/global-sidebar/hooks/use-global-sidebar.js]: From Eru's earlier PR 

```suggestion
import { isEnabled } from '@automattic/calypso-config';

export const useGlobalSidebar = ( siteId, sectionGroup ) => {
	let shouldShowGlobalSidebar = false;
	if ( isEnabled( 'layout/dotcom-nav-redesign' ) ) {
		shouldShowGlobalSidebar =
			sectionGroup === 'me' ||
			sectionGroup === 'reader' ||
			sectionGroup === 'sites-dashboard' ||
			( sectionGroup === 'sites' && ! siteId );
	}
	return { shouldShowGlobalSidebar };
};
```

Comment (fushar) [reply client/layout/global-sidebar/hooks/use-global-sidebar.js]: Did some smoke testing and I can confirm that adding the flag here does the job of gating all changes in this PR.

Comment (eoigal) [reply client/layout/global-sidebar/hooks/use-global-sidebar.js]: Committed, thanks for the suggestion 👍

**Comment by davemart-in** [comment]
I just did a review. Comparing to the [latest v4 HTML Prototype](https://wpcomnav4.web.app/
), here are the things that stand out:

- [ ] Increase width of global left nav from 272.px to 314px
- [x] Move notifications icon back up into the header of the global left nav (and remove it from the body of the global left nav)
- [x] Remove Reader from the body of the global left nav
- [x] Update the global left nav footer so it shows:

CleanShot 2024-02-13 at 08 22 19@2x

- [x] Add `cursor:pointer;` CSS to the help center (?) icon.
- [ ] Add tooltips to the search, notifications, reader, my profile, and help icons in the global left nav

- [ ] P2s and A8c Conversations appear to be overlapping for me in the reader section:

CleanShot 2024-02-13 at 08 26 57@2x

- [ ] We need to update the styles for the links in the global left nav:

CleanShot 2024-02-13 at 08 29 24@2x

Please let me know if I can assist with any of this. Thanks!

**Comment by ebinnion** [comment]
This PR is already 1,300+ lines of code added. Rather than continuing to make changes here by multiple people, let's 1) get confident that our feature flag is setup correctly, 2) ensure that the current state of the PR is reasonable, and 3) then launch. Assuming that our feature flag is correct, let's iterate on all of the smaller points in quick succession.

**Comment by eoigal** [comment]
https://github.com/Automattic/wp-calypso/pull/87628 is the last part of this PR to be merged. I think we can close this PR now.

**Pull request closed without merging**